src/import-data/ImportDataDestination.ts

```diff
@@ -111,7 +111,23 @@ export const ImportDataDestination = (props: ImportDataDestinationProps): ReactN
   const immediateImportTypes: ImportRequest['type'][] = ['tdr-snapshot-reference', 'catalog-snapshots'];
   const importMayTakeTime = !immediateImportTypes.includes(importRequest.type);
 
-  const { workspaces, refresh: refreshWorkspaces, loading: loadingWorkspaces } = useWorkspaces();
+  const {
+    workspaces,
+    refresh: refreshWorkspaces,
+    loading: loadingWorkspaces,
+  } = useWorkspaces([
+    'workspace.workspaceId',
```

Comment:
Just want to double check here, looks like your fields list is drifting considerably from the default list provided in `useWorkspaces` -- if I were doing this I'd be fearful of surprises in later logic from dropping params like `workspace.state`, `workspace.attributes`, etc.

I'm wondering if `useWorkspaces` should expose a `defaultWorkspaceFields` and then callers like this one can start with that and append stuff, eg: `[...defaultWorkspaceFields, 'workspace.workspaceId', <arg3>, <arg4>]`?

That said, I do appreciate _only asking for what we need_ so if my hesitation is uncalled for, by all means, go for it!

Reply:
On a similar line, `useWorkspace` hasn't yet been converted to typescript. When it is, will we be able to to type the return value `workspaces` so typescript knows which fields must be present or not based on the arguments? Depending on how many different common cases there are, we may want to create different APIs for each one, vs leaving the API open ended.

(I took a quick look and only saw one other case where the default fields weren't being used, so creating APIs for each custom use wouldn't be too bad.)

Reply:
Yeah, admittedly here I'm continuing to punt on what is a problem festering across the codebase: code is typed as if we always have the full Workspace object, but we almost always have some partial subset of it.

#4312 converts `useWorkspaces` to TypeScript, but I don't think it addresses the different return types.